packages/react-data/src/fetch/ReactDataFetch.js

```diff
@@ -44,50 +46,98 @@ function readResultMap(): Map<string, Result> {
   return map;
 }
 
-// TODO: options, auth, etc.
-export function fetch(url: string): Object {
-  const map = readResultMap();
-  const entry = map.get(url);
-  if (entry === undefined) {
-    let resolve = () => {};
-    const wakeable: Wakeable = new Promise(r => {
-      // TODO: should this be a plain thenable instead?
-      resolve = r;
-    });
-    const result: Result = {
-      status: Pending,
-      value: wakeable,
-    };
-    map.set(url, result);
-    const xhr = new XMLHttpRequest();
-    xhr.onload = function() {
-      // TODO: should we handle status codes?
-      if (result.status !== Pending) {
-        return;
+function toResult(thenable): Result {
+  const result: Result = {
+    status: Pending,
+    value: (null: any),
+  };
+  result.value = thenable.then(
```

Comment:
I tend do use the original thenable as the value instead of the return value. The first one will be held onto anyway but in theory the new one created by this one can be released. I'm not sure what's better though because some polyfills special cases a single listener. Not sure what native does.

Reply:
I think I was worried that React would be able to run before our continuation. But that's not possible because they run in order anyway, right? And we schedule ours first.

Reply:
I'll change it because my original concern doesn't make sense, so it should be ok. If not we'll learn something.